eknvfs/ekn-vfs.c

```diff
@@ -300,10 +316,22 @@ ekn_vfs_get_supported_uri_schemes (GVfs *self)
 static GFile *
 ekn_vfs_parse_name (GVfs *self, const char *parse_name)
 {
-  if (g_path_is_absolute (parse_name))
-    return g_vfs_parse_name (EKN_VFS_PRIVATE (self)->local, parse_name);
+  EknVfsPrivate *priv = EKN_VFS_PRIVATE (self);
+  GFile *retval = NULL;
+  gchar *scheme = g_uri_parse_scheme (parse_name);
+
+  gboolean is_supported_scheme =
+    g_strcmp0 (scheme, EKN_URI) == 0 ||
+    (priv->extensions && g_hash_table_contains (priv->extensions, scheme));
+
+  if (is_supported_scheme)
+    retval = ekn_vfs_get_file_for_uri (self, parse_name);
   else
-    return ekn_vfs_get_file_for_uri (self, parse_name);
+    retval = g_vfs_parse_name (EKN_VFS_PRIVATE (self)->local, parse_name);
```

Comment:
I don't know the difference between `g_vfs_parse_name` and `g_vfs_get_file_for_uri`, and how the "cascading" works in gvfs, but I think we could remove the repeated code for checking the supported scheme by doing something like:

```c
static GFile *
ekn_vfs_get_file_internal (GVfs *self, const char *uri)
{
  /* Code of current ekn_vfs_get_file_for_uri function */
  ...
  return retval;
}

static GFile *
ekn_vfs_get_file_for_uri (GVfs *self, const char *uri)
{
  GFile *retval = ekn_vfs_get_file_internal (self, uri);
  return retval ? retval : g_vfs_get_file_for_uri (EKN_VFS_PRIVATE (self)->local, uri);
}

static GFile *
ekn_vfs_parse_name (GVfs *self, const char *parse_name)
{
  GFile *retval = ekn_vfs_get_file_internal (self, parse_name);
  return retval ? retval : g_vfs_parse_name (EKN_VFS_PRIVATE (self)->local, parse_name);
}
```

Reply:
Mm, good call. I'll try that :)

Reply:
Okay, resolved with the newest commit. _Much_ better that way. I also added a comment to ekn_vfs_parse_name explaining why it's a smidge counter-intuitive.